src/components/ProductDetails/ProductDetails.tsx

```diff
@@ -0,0 +1,120 @@
+    async function fetchProducts() {
+      try {
+        const response = await fetch("api/products");
+        const data = await response.json();
+        setFetchedDetails(data[0]);
+        setIsLoading(false);
+      } catch (error) {
+        console.error(error);
+      }
+    }
+
+    fetchProducts();
+  }, []);
+
+  const handleIncrement = () => {
+    setQuantity((prevQuantity) => prevQuantity + 1);
+  };
+
+  const handleDecrement = () => {
+    setQuantity((prevQuantity) => (prevQuantity > 0 ? prevQuantity - 1 : prevQuantity));
+  };
+
+  const handleAddToCart = () => {
+    if (fetchedDetails && quantity > 0) {
+      addToCart({
+        ...fetchedDetails,
+        quantity,
+      });
+    }
+  };
+
+  function showTwoDecimals(num: number) {
+    return (Math.round(num * 100) / 100).toFixed(2);
+  }
+
+  if (isLoading) {
+    return <span className={styles.loader} />;
+  }
+
+  if (fetchedDetails) {
+    const {
+      company, title, description, price, discount,
+    } = fetchedDetails;
+
+    return (
+      <article className={styles.detailsContainer}>
+        <section className={styles.detailsText}>
+          <span className={styles.company}>{company}</span>
+          <h1 className={styles.title}>{title}</h1>
+          <p className={styles.description}>{description}</p>
+          <section className={styles.priceInfo}>
+            <div className={styles.priceAndDiscount}>
+              <span className={styles.discountedPrice}>
+                $
+                {showTwoDecimals(price * (1 - discount))}
+              </span>
+              <span className={styles.discount}>
+                {discount * 100}
+                %
+              </span>
+              {/* DELETE THIS - TEMP CODE FOR DEBUGGING */}
+              {cart.length > 0 ? (
+                <span className={styles.discount}>{cart[0].quantity}</span>
+              ) : null}
+              {/* DELETE THIS - TEMP CODE FOR DEBUGGING */}
+            </div>
+            <span className={styles.originalPrice}>
+              $
+              {showTwoDecimals(price)}
+            </span>
+          </section>
+        </section>
+        <section className={styles.utilities}>
+          <div className={styles.quantityControl}>
+            <button
+              type="button"
+              aria-label="Decrease quantity of product to be added to cart"
+              onClick={handleDecrement}
+            >
+              <MinusSvg />
+            </button>
+            <span>{quantity}</span>
+            <button
+              type="button"
+              aria-label="Increase quantity of product to be added to cart"
+              onClick={handleIncrement}
+            >
+              <PlusSvg />
+            </button>
+          </div>
+          <button
+            type="button"
+            className={styles.addToCartButton}
+            onClick={handleAddToCart}
+            aria-label="Add specified quantity of product to cart"
```

Comment:
What do you think if we include the quantity of the product?

Reply:
I think the quantity display above the add to cart button is sufficient. I added a modal for better user awareness.

Reply:
What about the user who uses text-to-speech? They won't know how much of the product they've added.